trio/_core/_parking_lot.py

```diff
@@ -98,13 +106,13 @@ class ParkingLot(metaclass=Final):
 
     # {task: None}, we just want a deque where we can quickly delete random
     # items
-    _parked = attr.ib(factory=OrderedDict, init=False)
+    _parked: OrderedDict[Task, None] = attr.ib(factory=OrderedDict, init=False)
```

Comment:
why not just change this to dict now?

Reply:
Why was it `'OrderedDict' in the first place? And why not just use `set`?

Reply:
That feels like it can be trivially be replaced with `set.pop()`? Or is there any difference in execution speed or something.

Reply:
Ah, so exactly as the comment says (duh) they want a FIFO/deque that also supports removing random items. Though given that items seem to only ever be removed randomly (in `_pop_several`) I'm a bit confused why it needs to be FIFO order.

Reply:
Err, I think you're confused a bit: the pop several method is ordered. The only reason we want fast random removal is because of cancellation!

Reply:
oooh. Thanks!